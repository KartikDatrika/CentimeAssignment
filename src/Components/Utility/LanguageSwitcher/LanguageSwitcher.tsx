import { getLanguageLocales } from '../../../Localization/locales';
import { useTranslation } from 'react-i18next';

export default function LanguageSwitcher() {
    const { i18n } = useTranslation();
    let languageLocales = getLanguageLocales();
    return (
        <div className="mb-5">
            <label className="columnName">Translate to</label>
            <select className="form-control" data-testid={"select"} onChange={(event) => i18n.changeLanguage(event.target.value)}>
                <option data-testid="select-option"key={"en"} value={"en"}>English</option>
                {Object.keys(languageLocales).map(ll => <option data-testid="select-option" key={ll} value={languageLocales[ll]}>{ll}</option>)}
            </select>
        </div>
    )
}
