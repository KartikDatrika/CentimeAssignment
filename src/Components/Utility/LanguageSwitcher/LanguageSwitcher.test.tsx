import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import LanguageSwitcher from './LanguageSwitcher';
import "./../../../Localization/i18n";

describe('Language Switcher Test Suite', () => {
    it('Language Switcher is rendering without any error !', () => {
        const { container } = render(<LanguageSwitcher/>);
        expect(container).toMatchSnapshot();
    });
    it('Options are available to change !', () => {
        const { getByTestId, getAllByTestId} = render(<LanguageSwitcher />);
        fireEvent.change(getByTestId("select"), {target: {value: 5}});
        let options = getAllByTestId("select-option");
        expect(options).toBeDefined();
    });
})
