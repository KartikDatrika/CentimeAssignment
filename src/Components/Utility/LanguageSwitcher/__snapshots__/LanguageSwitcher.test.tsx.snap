// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Language Switcher Test Suite Language Switcher is rendering without any error ! 1`] = `
<div>
  <div
    class="mb-5"
  >
    <label
      class="columnName"
    >
      Translate to
    </label>
    <select
      class="form-control"
      data-testid="select"
    >
      <option
        data-testid="select-option"
        value="en"
      >
        English
      </option>
      <option
        data-testid="select-option"
        value="af"
      >
        Afrikaans
      </option>
      <option
        data-testid="select-option"
        value="bg"
      >
        Bulgarian
      </option>
      <option
        data-testid="select-option"
        value="hr"
      >
        Croatian
      </option>
      <option
        data-testid="select-option"
        value="cs"
      >
        Czech
      </option>
      <option
        data-testid="select-option"
        value="da"
      >
        Danish
      </option>
      <option
        data-testid="select-option"
        value="nl"
      >
        Dutch
      </option>
      <option
        data-testid="select-option"
        value="fr"
      >
        French
      </option>
      <option
        data-testid="select-option"
        value="de"
      >
        German
      </option>
      <option
        data-testid="select-option"
        value="el"
      >
        Greek
      </option>
      <option
        data-testid="select-option"
        value="hi"
      >
        Hindi
      </option>
      <option
        data-testid="select-option"
        value="ja"
      >
        Japanese
      </option>
      <option
        data-testid="select-option"
        value="ko"
      >
        Korean
      </option>
      <option
        data-testid="select-option"
        value="ru"
      >
        Russian
      </option>
      <option
        data-testid="select-option"
        value="sr"
      >
        Serbian
      </option>
      <option
        data-testid="select-option"
        value="sk"
      >
        Slovak
      </option>
      <option
        data-testid="select-option"
        value="es"
      >
        Spanish
      </option>
      <option
        data-testid="select-option"
        value="sv"
      >
        Swedish
      </option>
      <option
        data-testid="select-option"
        value="th"
      >
        Thai
      </option>
      <option
        data-testid="select-option"
        value="tr"
      >
        Turkish
      </option>
      <option
        data-testid="select-option"
        value="vi"
      >
        Vietnamese
      </option>
    </select>
  </div>
</div>
`;
